crates/turbopack-core/src/resolve/plugin.rs

```diff
@@ -41,26 +41,37 @@ impl AfterResolvePluginCondition {
 
 /// A condition which determines if the hooks of a resolve plugin gets called.
 #[turbo_tasks::value]
-pub struct BeforeResolvePluginCondition {
-    glob: Vc<Glob>,
+pub enum BeforeResolvePluginCondition {
+    Request(Vc<Glob>),
+    Module(RcStr),
 }
 
 #[turbo_tasks::value_impl]
 impl BeforeResolvePluginCondition {
     #[turbo_tasks::function]
-    pub fn new(glob: Vc<Glob>) -> Vc<Self> {
-        BeforeResolvePluginCondition { glob }.cell()
+    pub fn from_request_glob(glob: Vc<Glob>) -> Vc<Self> {
+        BeforeResolvePluginCondition::Request(glob).cell()
+    }
+
+    #[turbo_tasks::function]
+    pub fn from_module(module: RcStr) -> Vc<Self> {
+        BeforeResolvePluginCondition::Module(module).cell()
     }
 
     #[turbo_tasks::function]
```

Comment:
Could we change that to an non-turbo-tasks function? To an normal async function.

This would avoid the need to spawn a new task to check the request.

Reply:
I think that was not clear enough... `matches` should be a normal function. `from_module` should stay a turbo-tasks function .